broker/src/main/java/io/camunda/zeebe/broker/partitioning/Partition.java

```diff
@@ -134,24 +136,31 @@ public ActorFuture<Partition> leave() {
           raftPartition
               .leave()
               .whenComplete(
-                  (leaveOk, leaveError) -> {
-                    concurrencyControl.run(
-                        () -> {
-                          if (leaveError != null) {
-                            result.completeExceptionally(leaveError);
-                            return;
-                          }
-                          concurrencyControl.runOnCompletion(
-                              startupProcess.shutdown(concurrencyControl, context),
-                              (shutdownOk, shutdownError) -> {
-                                if (shutdownError != null) {
-                                  result.completeExceptionally(shutdownError);
-                                  return;
-                                }
-                                result.complete(this);
-                              });
-                        });
-                  });
+                  (leaveOk, leaveError) ->
+                      concurrencyControl.run(
+                          () -> {
+                            if (leaveError != null) {
+                              result.completeExceptionally(leaveError);
+                              return;
+                            }
+                            concurrencyControl.runOnCompletion(
+                                startupProcess.shutdown(concurrencyControl, context),
+                                (shutdownOk, shutdownError) -> {
+                                  if (shutdownError != null) {
+                                    result.completeExceptionally(shutdownError);
+                                    return;
+                                  }
+                                  try {
+                                    FileUtil.deleteFolder(partitionDirectory);
```

Comment:
:upside_down_face: Cheap way to avoid issues with concurrent deletion, which should not happen but we also don't care too much (right?), use `FileUtil#deleteFolderIfExists`.